lib/nostr/nostr_api.dart

```diff
@@ -0,0 +1,145 @@
+      }
+    }
+    return null;
+  }
+
+  static Future<UserMetadata?> processRelays(
+      BuildContext context, ProfilePointer profile, String nip05Address) async {
+    String userDomain = _extractDomain(nip05Address);
+    const int metaData = 0;
+
+    for (String relayUrl in profile.relays ?? []) {
+      final relayDomain = _getDomainFromRelayUrl(relayUrl);
+      final formattedRelayDomain = relayToDomainMap[relayDomain] ?? relayDomain;
+      if (formattedRelayDomain == userDomain) {
+        final userDomainData = await _fetchInfoFromRelay(relayUrl, profile.pubkey, [metaData]);
+        if (userDomainData != null) {
+          return userDomainData;
+        }
+      }
+    }
+    await _showErrorDialog(context, noRelaysTitle, noUsersDomainRelay);
+
+    String? chosenRelayUrl = await _showRelayChoiceDialog(context, profile.relays ?? []);
+    if (chosenRelayUrl != null) {
+      final userData = await _fetchInfoFromRelay(chosenRelayUrl, profile.pubkey, [metaData]);
+      if (userData != null) {
+        return userData;
+      }
+    }
+
+    return null;
+  }
+
+  static Future<UserMetadata?> _fetchInfoFromRelay(
+      String relayUrl, String userPubKey, List<int> kinds) async {
+    try {
+      final relay = RelayApi(relayUrl: relayUrl);
+      final stream = await relay.connect();
+
+      relay.sub([
+        Filter(
+          kinds: kinds,
+          authors: [userPubKey],
+        )
+      ]);
+
+      await for (var message in stream) {
+        if (message.type == 'EVENT') {
+          final event = message.message as Event;
+
+          final eventContent = json.decode(event.content) as Map<String, dynamic>;
+
+          final userMetadata = UserMetadata.fromJson(eventContent);
+          relay.close();
+          return userMetadata;
+        }
+      }
+
+      relay.close();
+      return null;
+    } catch (e) {
+      print('[!] Error with relay $relayUrl: $e');
+      return null;
+    }
+  }
+
+  static Future<void> _showErrorDialog(
+      BuildContext context, String title, String errorMessage) async {
+    await showPopUp<void>(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertWithOneAction(
+          alertTitle: title,
+          alertContent: errorMessage,
+          buttonText: S.of(context).ok,
+          buttonAction: () => Navigator.of(context).pop(),
+        );
+      },
+    );
+  }
+
+  static String _extractDomain(String nip05Address) {
+    var parts = nip05Address.split('@');
+    return parts.length == 2 ? parts[1] : '';
+  }
+
+  static String _getDomainFromRelayUrl(String relayUrl) {
+    try {
+      var uri = Uri.parse(relayUrl);
+      return uri.host;
+    } catch (e) {
+      print('Error parsing URL: $e');
+      return '';
+    }
+  }
+
+  static Future<String?> _showRelayChoiceDialog(BuildContext context, List<String> relays) async {
+    String? selectedRelay;
+    relays.insert(0, 'Skip unrelated relays');
```

Comment:
same for translation.
but I don't see a use for this anyway!
if he taps on it, you return null and do nothing, if so remove it and he can close the popup easier and more understandable